components/shapediver/appbuilder/actions/AppBuilderActionSetBrowserLocationComponent.tsx

```diff
@@ -71,13 +72,7 @@ export default function AppBuilderActionSetBrowserLocationComponent(
 			);
 
 			// replace the value of the URL parameter "modelStateId" with the new value
-			if (modelStateId) {
-				newLocationUrl.searchParams.set(
-					QUERYPARAM_MODELSTATEID,
-					modelStateId,
-				);
-				newLocation = newLocationUrl.toString();
-			}
+			newLocation = applyModelStateToUrl(modelStateId, false).toString();
```

Comment:
@MajorMeerkatThe3rd this introduces a problem. The `newLocationUrl` is not the URL shown in the browser. It's a URL the browser will be redirected to. This change should be reverted.

Reply:
thank you!